staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -229,6 +229,9 @@ func (o AnnotateOptions) Validate() error {
 		if cmdutil.IsFilenameSliceEmpty(o.Filenames, o.Kustomize) {
 			return fmt.Errorf("one or more files must be specified as -f rsrc.yaml or --filename=rsrc.json")
 		}
+		if o.dryRunStrategy == cmdutil.DryRunServer {
```

Comment:
Hmm, I'm curious why you put it at the end rather than the beginning?

Reply:
Ah, moved up to return a biit faster.

These checks are for validation and they're fast.

Reply:
I don't really care about speed, I'm concerned about UX, since we're not showing all the possible errors (which is probably wrong). I'm curious if we could aggregate them somehow?

Reply:
Ahhhh I see.

`set image` here happens to be an example of a command w/ a `Validate` method that aggregates errors if we want to do that for more of the commands.